### PR TITLE
Run2-hcx188 First attempt to convert a few Hcal related algorithms to DD4HEP format

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: DetectorDescription/DDCMS/plugins/DDHCalAngular.cc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			       << "\tChild " << child.name() 
+			       << " NameSpace " << ns.name();
+#endif
+  double theta  = 90._deg;
+  int    copy   = startCopyNo;
+  double phi    = startAngle;
+  for (int ii=0; ii<n; ++ii) {
+    double phix   = ConvertTo( phi, deg );
+    double phiy   = phix + 90._deg;
+    int    iphi   = (phix > 0) ? int(phix+0.1) : int(phix-0.1);
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/plugins/DDHCalAngular.cc]: Is rounding phix to the nearest degree actually necessary? I've seen and changed code like this previously, and my conclusion was that rounding to the nearest degree in general was not needed.

File: DetectorDescription/DDCMS/plugins/DDHCalAngular.cc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    if (iphi >= 360) iphi -= 360;
+    phix = iphi;
+    dd4hep::Rotation3D rotation;
+    if (iphi != 0) {
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/plugins/DDHCalAngular.cc]: For lines 44-45 and 48, you can see in PR #24509 how I handled nearly identical code. My approach in this case would be:
 if (phix >= 2._pi) phix -= 2._pi; for lines 44-45
 if (std::abs(phix) >= 1.0_deg) { for line 48

Is it important that your version rounds a phix of 0.9 degree to 1.0 degree, while my version considers 0.9 degree as being 0?

File: DetectorDescription/DDCMS/plugins/DDHCalAngular.cc (hunk 1 above)
Comment (bsunanda) [reply DetectorDescription/DDCMS/plugins/DDHCalAngular.cc]: Did that changes

File: DetectorDescription/DDCMS/plugins/DDHCalAngular.cc (hunk 2 above)
Comment (cvuosalo) [reply DetectorDescription/DDCMS/plugins/DDHCalAngular.cc]: In the new version, the reduction of phix if it is greater than 2 pi has been omitted. It needs to be added, as shown in my previous comment. Also, iphi is unnecessary and potentially confusing.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**